comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -420,6 +423,12 @@ func (r *secretResolver) Refresh() (string, error) {
 
 	// when Refreshing secrets, only update what the allowlist allows by passing `true`
 	refreshResult := r.processSecretResponse(secretResponse, true)
+	// add the results to the audit file, if filename is non-empty
+	if r.auditFilename != "" {
+		if err := AddToRefreshAuditFile(r.auditFilename, secretResponse, r.origin); err != nil {
```

Comment:
We should refresh the audit file only the value of the secret changed not every time we fetch secret again.

Reply:
Done, checking that a non-zero number of handles were refreshed by `processSecretResponse` does this.